File: Ride.js
const mongoose = require('mongoose');

const rideSchema = new mongoose.Schema({
  passenger: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  driver: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  origin: { lat: Number, lng: Number, name: String },
  destination: { lat: Number, lng: Number, name: String },
  status: {
    type: String,
    enum: ['SEARCHING', 'MATCHED', 'DRIVER_ARRIVING', 'IN_PROGRESS', 'COMPLETED', 'CANCELLED'],
    default: 'SEARCHING',
  },
  fare: { type: Number, default: 0 },
  distanceKm: Number,
  durationMin: Number,
  rating: { byPassenger: Number, byDriver: Number },
  paymentStatus: { type: String, enum: ['PENDING', 'PAID'], default: 'PENDING' },
  razorpayOrderId: String,
}, { timestamps: true });

module.exports = mongoose.model('Ride', rideSchema);

File: Ride.kt
package com.rideshare.model

import com.google.android.gms.maps.model.LatLng

data class Ride(
    val id: String = "",
    val driverId: String = "",
    val passengerId: String = "",
    val origin: LatLng = LatLng(0.0, 0.0),
    val destination: LatLng = LatLng(0.0, 0.0),
    val originName: String = "",
    val destinationName: String = "",
    val status: RideStatus = RideStatus.SEARCHING,
    val fare: Double = 0.0,
    val distanceKm: Double = 0.0,
    val durationMin: Int = 0,
    val driverName: String = "",
    val driverRating: Float = 0f,
    val vehicleNumber: String = "",
    val createdAt: Long = System.currentTimeMillis(),
)

enum class RideStatus { SEARCHING, MATCHED, DRIVER_ARRIVING, IN_PROGRESS, COMPLETED, CANCELLED }
